### PR TITLE
[block] deprecate some code in block.go and body.go

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
as title

Fixes #(issue)

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: consensus/scheme/rolldpos/subchain_test.go [hunk 1]
```diff
@@ -1,73 +0,0 @@
...
-	"github.com/stretchr/testify/require"
-	"google.golang.org/protobuf/types/known/timestamppb"
-
-	"github.com/iotexproject/iotex-core/action"
-	"github.com/iotexproject/iotex-core/blockchain/block"
-	"github.com/iotexproject/iotex-core/pkg/version"
-	"github.com/iotexproject/iotex-core/test/identityset"
-)
-
-func TestPutBlockToParentChain(t *testing.T) {
```
Comment (dustinxie) [inline consensus/scheme/rolldpos/subchain_test.go:23]: this test is no longer valid after subchain feature/code was removed

File: consensus/scheme/rolldpos/subchain_test.go [hunk 2]
```diff
@@ -1,73 +0,0 @@
...
-						Version: version.ProtocolVersion,
-						Nonce:   102,
-					},
-					SenderPubKey: pubKey.Bytes(),
-					Signature:    action.ValidSig,
-				},
-			},
-		},
-	}
-	require.NoError(t, blk.ConvertFromBlockPb(blkpb, 0))
```
Comment (dustinxie) [inline consensus/scheme/rolldpos/subchain_test.go:67]: here only tests block conversion from protobuf, which exist in test code under /blockchain/block/

File: blockchain/block/block_deserializer.go [hunk 3]
```diff
@@ -32,11 +34,14 @@ func (bd *Deserializer) SetEvmNetworkID(id uint32) *Deserializer {
...
+		err error
+	)
+	if err = b.Header.LoadFromBlockHeaderProto(pbBlock.GetHeader()); err != nil {
 		return nil, errors.Wrap(err, "failed to deserialize block header")
 	}
-	if err := b.Body.LoadProto(pbBlock.GetBody(), bd.evmNetworkID); err != nil {
+	if b.Body, err = bd.fromBodyProto(pbBlock.GetBody()); err != nil {
 		return nil, errors.Wrap(err, "failed to deserialize block body")
 	}
 	if err := b.ConvertFromBlockFooterPb(pbBlock.GetFooter()); err != nil {
```
Comment (Liuhaai) [inline blockchain/block/block_deserializer.go:47]: err = xxx

File: blockchain/block/block_deserializer.go [hunk 4]
```diff
@@ -77,5 +86,6 @@ func (bd *Deserializer) DeserializeBody(buf []byte) (*Body, error) {
 	if err := proto.Unmarshal(buf, &pb); err != nil {
 		return nil, errors.Wrap(err, "failed to unmarshal block body")
 	}
-	return bd.FromBodyProto(&pb)
+	b, err := bd.fromBodyProto(&pb)
```
Comment (Liuhaai) [inline blockchain/block/block_deserializer.go:89]: b, err := balabala
if err != nil {
return nil, err
}
return &b, nil

**Review by Liuhaai — APPROVED** [review]

File: blockchain/block/block_deserializer.go (hunk 4 above)
Comment (dustinxie) [reply blockchain/block/block_deserializer.go:89]: fixed

File: blockchain/block/block_deserializer.go (hunk 3 above)
Comment (dustinxie) [reply blockchain/block/block_deserializer.go]: fixed

**Review by raullenchai — APPROVED** [review]

**Pull request merged**